riscemu/parser.py

```diff
@@ -112,6 +113,28 @@ def take_arguments(tokens: Peekable[Token]) -> Iterable[str]:
         # raise ParseException("Expected newline, instead got {}".format(tokens.peek()))
 
 
+def parse_instruction_arguments(args: Tuple[str]) -> Tuple[str]:
+    """
+    Parses argument Tuples of instructions. In this process canonicalize register names
+    :param args: A tuple of an instructions arguments
+    :return: A tuple of an instructions parsed arguments
+    """
+    return tuple(canonicalize_register_names(args))
+
+
+def canonicalize_register_names(args: Tuple[str]) -> Iterable[str]:
+    """
+    Canonicalizes register names.
+    :param args: A tuple of an instructions arguments
+    :return: An iterator over the arguments of an instruction, but with canonicalized register names
+    """
+    for arg in args:
+        if arg in REG_NAME_CANONICALIZER.keys():
+            yield REG_NAME_CANONICALIZER[arg]
+        else:
+            yield arg
```

Comment:
I can't really speak to how this would best fit with the overall architecture of the framework, but I'd recommend the following changes for this function specifically:
```suggestion
def register_name_for_indexed(indexed_name: str) -> str:
    """
    Canonicalizes register name.
    :param args: A tuple of an instructions arguments.
    :return: An iterator over the arguments of an instruction, but with canonicalized register names.
    
    
    """
    return REG_NAME_CANONICALIZER.get(arg, arg)
```

The function name describes a bit better what it does, I would have expected the canonical name to be the indexed name. The function operates one element at a time, the client can decide whether to get just one name or a tuple or a list at the callsite (`tuple(register_name_for_indexed(arg) for arg in args))`). There's a built-in method for the pattern of getting a value with a default if it's not in the dictionary.

Reply:
oh and I think some examples in the docstring would be good

Reply:
Almost @superlopuh , I think you are missing that `args` is an array here.

But still, I'd simplify the `if .. in ...` and item getting into a single `.get(arg, default)` call.